pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -715,6 +673,45 @@ void registerBackendAgnosticStateVectorMethods(PyClass &pyclass) {
     pyclass.def("size", &StateVectorT::getLength);
 }
 
+/**
+ * @brief Register controlled gate operations for a statevector.
+ *
+ * @tparam StateVectorT State vector type
+ * @tparam PyClass Nanobind class type
+ * @param pyclass Nanobind class to bind methods to
+ */
+template <class StateVectorT, class PyClass>
+void registerControlledGates(PyClass &pyclass) {
```

Comment:
Small thing: for organization, would it be better to move this to be right underneath `registerGatesForStatevector()`